File: frontend/src/types/oauth.ts
export interface OAuthProviderRead {
  label: string;
  provider: string;
}

export interface OAuthAuthorizationStartRead {
  authorization_url: string;
  state: string;
}

export interface OAuthBindingRead {
  created_at: string;
  email: string | null;
  expires_at: string | null;
  external_id: string;
  id: string;
  provider: string;
  updated_at: string;
  user_id: string;
}

export interface OAuthCallbackPayload {
  code: string;
  provider: string;
  state: string;
}

export interface OAuthBindPayload {
  code: string;
  state: string;
}

export interface OAuthUnbindResult {
  unbound: boolean;
}

File: backend/app/schemas/oauth.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class OAuthBindingRead(BaseModel):
    id: UUID
    user_id: UUID
    provider: str
    external_id: str
    email: str | None
    expires_at: datetime | None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class OAuthAuthorizationStartRead(BaseModel):
    authorization_url: str
    state: str
